added zip function in textbase cli tool

jahanvir opened the pull request:
This is an extension of previous pull request

https://github.com/cofactoryai/textbase/pull/97

## Scope
create zip file of bot using textbase CLI tool

Example:
`poetry run python textbase/textbase_cli.py compress
`
`textbase-client compress`


- [x] `[Sub task]`


### Screenshots
---
![image](https://github.com/cofactoryai/textbase/assets/30360093/3826e215-55d4-43bf-ae87-528ecd584999)


### Developer checklist
- [x] I’ve manually tested that code works locally on desktop and mobile browsers.
- [x] I’ve reviewed my code.
- [x] I’ve removed all my personal credentials (API keys etc.) from the code.

bchhabra2490 commented on textbase/textbase_cli.py:
```diff
@@ -35,6 +37,25 @@ def test(path):
+def compress():
+    click.echo(click.style(f"Creating zip file for deployment", fg='green'))
+
+    files_to_zip = ['requirements.txt', 'main.py']
```
there can be other folders, or files users may want to zip. Better will be to zip everything in the folder from which we run this command. That will avoid a lot of dir problems we faced today.

sammyCofactory commented:
Hey @jahanvir, thanks for contributing to textbase! Now, there are a few issues with your code:
1. If the files `main.py` and `requirements.txt` don't exist, there is still an empty zip archive being made. The error message which is shown is sufficient but the execution is not being stopped there.
2. Adding a `--path` flag will be really helpful as currently the users have no choice but to always put their `main.py` and `requirements.txt` inside the `textbase` folder. The way this flag will work is, it'll first check if those two main files are present in the pwd:
   - If they are, then zip everything in the pwd.
   - If they are not, then tell the user to point it to a folder which does.

jahanvir commented:
Hi @sammyCofactory,

I got your point.
I have made changes as per feedback.

1. Added path option for main and requirement filed.
2. Also added the function to check if filed exist on the given path or not so that it does not create an empty deploy.zip file.

<img width="1110" alt="image" src="https://github.com/cofactoryai/textbase/assets/30360093/10974af0-eeb7-405f-8899-59bd44457c28">

sammyCofactory commented:
@jahanvir, took a look at your updated code. While you have added the `--path` parameter, it is not quite as we want it to be. I think you could modify it so there is only one `--path` parameter and not `--main_path` and `--requirements_path`. The way this will work has been explained in my previous comment (I have edited it for better understanding). Doing so will also remove the requirement for the extra `fileExist` function.

Do ask if you have any doubts.

jahanvir commented:
Hi @sammyCofactory, so in this case there would be no need to show error. User would be sharing the correct path. And both main.py and requirements.txt in one folder only.

sammyCofactory commented:
Well, we should still have an error message. What if the user shares the wrong path?

Yes, the user has to give a path to a folder where he has both the files. If he doesn't, the default fallback will be to the current folder.

jahanvir commented:
Okay, @sammyCofactory 

Just to confirm user will enter the path of directory not the complete path of main.py or requirements.txt right ??

so suppose the case is


> folder1/folder2/main.py

User will enter folder1/folder2

and then check needs to be run for 

> path + main.py and path + requirements.txt

sammyCofactory commented:
Yes, exactly! Also, when the files are to be zipped up, zip all the contents of the folder path which the user gave, i.e., zip all the contents of `folder2`.

jahanvir commented:
@sammyCofactory  Okay cool

I have made changes so that it will 

- take a directory path.
- check for main.py and requirements.txt, raise error in case its not present and stop execution there
- compress everything present in the directory into deploy.zip

Sample test case
<img width="271" alt="Screenshot 2023-09-07 at 8 11 57 PM" src="https://github.com/cofactoryai/textbase/assets/30360093/f74e32d8-28eb-491b-a591-7a12bcc06278">
p execution.

Sample deply.zip file
<img width="782" alt="image" src="https://github.com/cofactoryai/textbase/assets/30360093/a7a7afc0-0edb-4c7c-a3ec-ea22f8b5e9bc">

In case main.py is not present in directory,
<img width="961" alt="image" src="https://github.com/cofactoryai/textbase/assets/30360093/7867b4fe-3bfe-4ae6-ae44-1621ba550e62">

sammyCofactory commented:
@jahanvir, there is one small thing I observed while testing out your code. There is a `deploy.zip` being created inside `deploy.zip`. Do run the code yourself and open `deploy.zip` if this seems too confusing. Can you add a check in the loop to avoid this?

jahanvir commented:
@sammyCofactory yeah that's happening when directory path is root directory as newly created deploy.zip will be in the same directory too.

I have added the condition in loop to avoid that!!

sammyCofactory reviewed (approved):
Went through your changes and tested out your code. LGTM!
